File: frontend/src/app/domain/responses/playlist-response.ts
import {ExternalURLObject} from '../objects/external-urlobject';
import {ImageObject} from '../objects/image-object';
import { OwnerObject } from "../objects/owner-object";
import {TracksObject} from '../objects/tracks-object';

export interface PlaylistResponse {
  collaborative: boolean;
  description: string;
  external_urls: ExternalURLObject;
  href: string;
  id: string;
  images: ImageObject[];
  name: string;
  owner: OwnerObject;
  public: boolean;
  snapshot_id: string;
  tracks: TracksObject;
  type: string;
  uri: string;
}

File: backend/src/main/java/organizer/backend/domain/responses/PlaylistResponse.java
package organizer.backend.domain.responses;

import com.fasterxml.jackson.annotation.JsonProperty;
import organizer.backend.domain.objects.ExternalURLObject;
import organizer.backend.domain.objects.ImageObject;
import organizer.backend.domain.objects.OwnerObject;
import organizer.backend.domain.objects.TracksObject;

public record PlaylistResponse(Boolean collaborative,
                               String description,
                               ExternalURLObject external_urls,
                               String href,
                               String id,
                               ImageObject[] images,
                               String name,
                               OwnerObject owner,
                               @JsonProperty("public") Boolean publicValue,
                               String snapshot_id,
                               TracksObject tracks,
                               String type,
                               String uri
                               ) {
}
